src/status_im/contexts/preview/quo/wallet/token_input.cljs

```diff
@@ -20,39 +25,72 @@
+                             :networks            networks
+                             :title               title
+                             :customization-color :blue
+                             :show-keyboard?      false
+                             :allow-selection?    true
+                             :crypto?             true})]
     (fn []
-      [preview/preview-container
-       {:state                     state
-        :descriptor                descriptor
-        :full-screen?              true
-        :component-container-style {:flex            1
-                                    :justify-content :space-between}}
-       [quo/token-input (assoc @state :value @value)]
-       [quo/numbered-keyboard
-        {:container-style {:padding-bottom (safe-area/get-top)}
-         :left-action     :dot
-         :delete-key?     true
-         :on-press        set-value
-         :on-delete       delete}]])))
+      (let [{:keys [currency token conversion-rate
+                    crypto?]}             @state
+            [input-state set-input-state] (rn/use-state controlled-input/init-state)
+            input-amount                  (controlled-input/input-value input-state)
+            swap-between-fiat-and-crypto  (fn []
+                                            (set-input-state
+                                             (fn [input-state]
+                                               (controlled-input/set-input-value
+                                                input-state
+                                                (let [new-value
+                                                      (if (not crypto?)
+                                                        (utils/cut-crypto-decimals-to-fit-usd-cents
+                                                         conversion-rate
+                                                         (money/fiat->crypto input-amount
+                                                                             conversion-rate))
+                                                        (utils/cut-fiat-balance-to-two-decimals
+                                                         (money/crypto->fiat input-amount
+                                                                             conversion-rate)))]
+                                                  (number/remove-trailing-zeroes
+                                                   new-value))))))
+            converted-value               (if crypto?
+                                            (utils/prettify-balance currency
+                                                                    (money/crypto->fiat input-amount
+                                                                                        conversion-rate))
+                                            (utils/prettify-crypto-balance
+                                             (or (clj->js token) "")
```

Comment:
I'm wondering why do we convert the `token` data to JS, why do we need it?

Reply:
this is actually a way of avoiding `(name token)` that could throw an exception